lib/kubernetes-deploy/kubernetes_resource/replica_set.rb

```diff
@@ -53,7 +61,8 @@ def exists?
     def fetch_events
       own_events = super
       return own_events unless @pods.present?
-      own_events.merge(@pods.first.fetch_events)
+      most_useful_pod = @pods.find(&:deploy_failed?) || @pods.find(&:deploy_timed_out?) || @pods.first
```

Comment:
so you are trying to find a pod that has explicitly been identified as the one that cause the deploy to fail, falling back to one that explicitly caused a timeout, falling back to just the first pod (arbitrarily)?